### DISCUSSION TITLE
Blockly Script

### DISCUSSION THREAD

**Discussion by ledermann** [solectrus/solectrus#2069] [Show and tell]
Der Ordnung halber verschiebe ich eine Diskussion hierher:

@StevieC121176 schreibt in https://github.com/solectrus/solectrus/issues/2064#issuecomment-1578230618:

> Status ist bei mir jetzt auch nicht mehr nur "Live" und "Fail" Ich schreibe jetzt meine Wechselrichter Statusmeldungen per Blockly Script in einen Datenpunkt der durch MQTT published wird und ich mir den dann in Solectrus anzeigen lasse.

@spirit0607 schreibt in https://github.com/solectrus/solectrus/issues/2064#issuecomment-1582060741:

> Könntest du mir das Blockly Script zur Verfügung stellen? Oder könnte man das auch evtl. mit in die Installationsanleitung MQTT als "optional" mit aufnehmen?? Das wäre super. Danke! PS: das müsste doch auch für die Case Temperatur gehen, oder?

**Comment by StevieC121176** [comment]
Kann ich gerne machen. 
Es ist so, der Plenticore Adapter für meine WR liefert als Statusmeldungen nur Zahlen, die dann intern im WR als Textmeldungen hinterlegt sind. Um diese zu nutzen habe ich mir in ioBroker einen leeren Datenpunkt angelegt.
Das Blockly Script wandelt nun die Zahlen in Textmeldungen um und schreibt die aktuelle immer in den Datenpunkt. Den lasse ich durch den MQTT Client vom iOBroker publischen und der MQTT Collector von Solectrus tut dann sein übriges.

Bei der Batterietemperatur fällt ein Schritt weg. Der BYD Adapter vom ioBroker stellt die Temperatur zur Verfügung. Diese muss nur vom MQTT Client published werden.

Um mein Setup zu nutzen müsst du also den ioBroker haben, einen Kostal Plenticore WR und eine BYD Batterie inkl. installierter Adapter für ioBroker. (javascript, Mqtt Client, MQTT Broker, Plenticore und BYDHVS)

**Comment by spirit0607** [comment]
Ah alles klar. Danke für die Rückmeldung. Stimmt, das ist in der Tat je nach Setup schon bei jedem sehr individuell. Ich werde mal probieren, den aktuellen Status per iobroker und MQTT Client zu pushen.

**Reply by StevieC121176** [reply]
Ja das ist die einfachste Methode. Wenn dein WR Adapter oder der Battaerieadapter einen Datenpunkt für die Statusmeldungen zur Verfügung stellt, dann kannst du über den MQTT Client diese einzelnen Datenpunkte für Solectrus bereit stellen
